packages/react-dom/src/events/ReactDOMEventListener.js

```diff
@@ -219,6 +230,29 @@ export function dispatchEvent(
     clearIfContinuousEvent(domEventName, nativeEvent);
   }
 
+  if (
+    enableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay &&
+    enableSelectiveHydration &&
+    isCapturePhaseSynchronouslyHydratableEvent(domEventName)
+  ) {
```

Comment:
There's nothing that checks whether this is capture/bubble phase so this block will be executed twice per event when it only needs to be in the capture phase. If it wasn't able to hydrate, it might not even get there when you add stopPropagation, but if was able to hydrate it'll check again unnecessarily.

Reply:
Added it in this follow up: #22502